File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        HashMap<String, Integer> sf = new HashMap<>();
        for (String str : arr) {
            if (sf.containsKey(str)) {
                sf.put(str, sf.get(str) + 1);
            } else {
                sf.put(str, 1);
            }
        }
        int i = 0;
        for (String s : arr) {
            if (sf.get(s) == 1) {
                ++i;
            }
            if (i == k) {
                return s;
            }
        }
        return "";
    }
}

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string, int> sf; // string freq
        for (auto str : arr) {
            sf[str]++;
        }
        int i = 0;
        for (auto it : arr) { // arr preserve the order
            if (sf[it] == 1) {
                ++i;
            }
            if (i == k) {
                return it;
            }
        }
        return "";
    }
};